data/lm/generate_lm.py

```diff
@@ -191,17 +222,37 @@ def main():
     )
     parser.add_argument(
         "--discount_fallback",
-        help="To try when such message is returned by kenlm: 'Could not calculate Kneser-Ney discounts [...] rerun with --discount_fallback'",
+        help="To try when such message is returned by kenlm: 'Could not calculate Kneser-Ney "
+             "discounts [...] rerun with --discount_fallback'",
         action="store_true",
     )
 
     args = parser.parse_args()
 
+    # Basic checks on user supplied arguments
+    if args.kenlm_bins == "path/to/kenlm/build/bin/":
+        parser.error("Update kenlm_bins from documentation value to actual path for kenlm")
+
+    if not os.path.isdir(args.kenlm_bins):
+        parser.error("kenlm_bins must be a valid directory")
+
+    kenlm_files = ["lmplz","filter","build_binary"]
+    for f in kenlm_files:
+        if not os.path.isfile(os.path.join(args.kenlm_bins, f)):
+            parser.error(f"Required kenlm file {f} not found in kenlm_bins directory")
+
+    if len(args.top_k) > len(args.input_txt):
+        parser.error("Number of top_k arguments passed should not exceed number of input_txt"
+                     " arguments passed")
+
+    if not os.path.isdir(args.output_dir):
+        parser.error("output_dir must be a valid directory")
+
     data_lower, vocab_str = convert_and_filter_topk(args)
```

Comment:
It's not really your fault, but this is making me wonder whether we are not memory hungry for nothing here, with `vocab_str` and maybe we should directly write to a file that will be passed to `build_lm` instead?

Reply:
I agree it probably could be made more memory efficient rather than holding the vocab string in memory all at once. I did casually look at my memory use when running but it didn't cause me issues -the machine has 64Gb RAM - but making it so it can still run over a large corpus on a lower resource machine seems wise. Will see what I can do here too.

Reply:
Yep, I have the same isssue, my working desktop has 64 CPUs, 128 threads and 128GB of RAM.